extensions-core/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcStructConverter.java

```diff
@@ -0,0 +1,198 @@
+    }
+    return new ArrayList<Object>(orcList);
+  }
+
+  static Map<Object, Object> convertMap(
+      TypeDescription fieldDescription,
+      OrcMap<? extends WritableComparable, ? extends WritableComparable> map
+  )
+  {
+    Map<Object, Object> converted = new HashMap<>();
+    TypeDescription keyDescription = fieldDescription.getChildren().get(0);
+    TypeDescription valueDescription = fieldDescription.getChildren().get(1);
+    for (WritableComparable key : map.navigableKeySet()) {
+      Object newKey = convertPrimitive(keyDescription, key);
+      if (valueDescription.getCategory().isPrimitive()) {
+        converted.put(newKey, convertPrimitive(valueDescription, map.get(key)));
+      } else {
+        converted.put(newKey, map.get(key));
+      }
+    }
+    return converted;
+  }
+
+  private Map<String, Object> convertMap(OrcStruct map)
+  {
+    Map<String, Object> converted = new HashMap<>();
+    for (String key : map.getSchema().getFieldNames()) {
+      converted.put(key, convertField(map, key));
+    }
+    return converted;
+  }
+
+  @SuppressForbidden(reason = "new DateTime(java.util.Date)")
+  private static Object convertPrimitive(TypeDescription fieldDescription, WritableComparable field)
+  {
+    /*
+        ORC TYPE    WRITABLE TYPE
+        binary      org.apache.hadoop.io.BytesWritable
+        bigint      org.apache.hadoop.io.LongWritable
+        boolean     org.apache.hadoop.io.BooleanWritable
+        char        org.apache.hadoop.io.Text
+        date        org.apache.hadoop.hive.serde2.io.DateWritable
+        decimal     org.apache.hadoop.hive.serde2.io.HiveDecimalWritable
+        double      org.apache.hadoop.io.DoubleWritable
+        float       org.apache.hadoop.io.FloatWritable
+        int         org.apache.hadoop.io.IntWritable
+        smallint    org.apache.hadoop.io.ShortWritable
+        string      org.apache.hadoop.io.Text
+        timestamp   org.apache.orc.mapred.OrcTimestamp
+        tinyint     org.apache.hadoop.io.ByteWritable
+        varchar     org.apache.hadoop.io.Text
+     */
+    switch (fieldDescription.getCategory()) {
+      case STRING:
+      case CHAR:
+      case VARCHAR:
+        return ((Text) field).toString();
+      case BOOLEAN:
+        return ((BooleanWritable) field).get();
+      case BYTE:
+        return ((ByteWritable) field).get();
+      case SHORT:
+        return ((ShortWritable) field).get();
+      case INT:
+        return ((IntWritable) field).get();
+      case LONG:
+        return ((LongWritable) field).get();
+      case FLOAT:
+        return ((FloatWritable) field).get();
+      case DOUBLE:
+        return ((DoubleWritable) field).get();
+      case DECIMAL:
+        return ((HiveDecimalWritable) field).getHiveDecimal().doubleValue();
+      case TIMESTAMP:
+        return ((OrcTimestamp) field).getTime();
+      case DATE:
+        // todo: is the the best way to go from java.util.Date to DateTime?
```

Comment:
Please don't commit TODOs. Also, the best way is probably `DateTimes.utc(theJavaUtilDate.getTime())`.